### ISSUE TITLE
Heading wrapped inside two Texts

### ISSUE THREAD

**Issue by megacherry** [gmsgowtham/react-native-marked#522]
Hi @gmsgowtham,

first of all: Thanks for your awesome work. Then I have a question regarding the string "## Heading2". It seems that it is wrapped in two text elements. That means if I define a custom overwrite (heading) that already contains a style it will be overwritten by the inner one.

Can you reproduce this?

```
<Text selectable={true} style={{"borderBottomColor": "#d0d7de", "borderBottomWidth": 1, "color": "#333333", "fontSize": 28, "fontWeight": "500", "lineHeight": 36, "marginVertical": 8, "paddingBottom": 4}}>
<Text selectable={true} style={{"borderBottomColor": "#d0d7de", "borderBottomWidth": 1, "color": "#333333", "fontSize": 28, "fontWeight": "500", "lineHeight": 36, "marginVertical": 8, "paddingBottom": 4}}>
Heading2
</Text>
</Text>
```

Thanks in advance.

Best regards,
Andy

**Comment by gmsgowtham** [comment]
@megacherry Can you please provide a reproduction?

**Comment by megacherry** [comment]
[](https://github.com/megacherry/repro-react-native-marked)

If you ran yarn ios, you will see something like this:

```
<Heading style={{"borderBottomColor": "#d0d7de", "borderBottomWidth": 1, "color": "#333333", "fontSize": 28, "fontWeight": "500", "lineHeight": 36, "marginVertical": 8, "paddingBottom": 4}}>
<Paragraph style={{"borderBottomColor": "#d0d7de", "borderBottomWidth": 1, "color": "#333333", "fontSize": 28, "fontWeight": "500", "lineHeight": 36, "marginVertical": 8, "paddingBottom": 4}}>
Heading2
</Paragraph>
</Heading>
```

I expected it to render only <Heading>.

**Comment by gmsgowtham** [comment]
@megacherry This is an interesting issue. I'll see how it can be solved. 


> Some background:

[Marked.js](https://github.com/markedjs/marked) outputs the following lexer data for the markdown string `## Hello world`

```js
[
  {
    type: "heading", raw: "## Hello world", depth: 2, text: "Hello world", tokens: [
      { type: "text", raw: "Hello world", text: "Hello world" }
    ]
  }
]
```

The parser tries to parse both the top level token and the child tokens which results in the output you mentioned.

**Comment by megacherry** [comment]
Ah, thanks for the info. I already searched your lib and didn't find a solution.

Please contact me if you want me to test something else or if I can help you further.

**Comment by gmsgowtham** [comment]
@megacherry This is resolved in v6.0.0

**Comment by gmsgowtham** [comment]
There still could be instances of duplicate text wrapping. I'll try to iron out those.
